model-optimizer/extensions/ops/If.py

```diff
@@ -0,0 +1,321 @@
+        internal_result['output_id'] = if_output_port_idx
+
+    @staticmethod
+    def update_body_parameters_type(if_node: Node, condition: bool):
+        """
+        Update the data type for If body Parameter nodes based on data type of the outer graph nodes producing data
+        for them.
+
+        :param if_node: The If node
+        :param condition: the boolean defining a condition (then/else) graph
+        :return: None
+        """
+        assert if_node.soft_get('type') == 'If'
+
+        subgraph = if_node.then_graph if condition else if_node.else_graph
+        for node in subgraph.get_op_nodes():
+            if node.has('input_id'):
+                assert node.soft_get('type') == 'Parameter'
+                input_port_id = node['input_id']
+                input_type = if_node.in_port(input_port_id).get_data_type()
+                node.data_type = input_type
+                log.debug('Updated data type for the body node with name "{}" with value {}'
+                          .format(node.name, node.data_type))
+
+    @staticmethod
+    def update_body_parameters_shape(if_node: Node, condition: bool):
+        """
+        Update shape for If body parameters.
+
+        :param if_node: The If node
+        :param condition: the boolean defining a condition (then/else) graph to add connect the body
+        :return: None
+        """
+        subgraph = if_node.then_graph if condition else if_node.else_graph
+        for node in subgraph.get_op_nodes():
+            if node.has('input_id'):
+                assert node.soft_get('type') == 'Parameter'
+                input_port_id = node['input_id']
+                input_shape = if_node.in_port(input_port_id).data.get_shape()
+                if node.soft_get('shape', None) is None:
+                    node['shape'] = None
+                node.shape = input_shape.copy()
+                log.debug('Updated shape for the body node with name "{}" with value {}'
+                          .format(node.soft_get('name', node.soft_get('id')), node.shape))
+
+    @staticmethod
+    def update_if_output_ports_shape(if_node: Node):
+        """
+        Update shape and values for If output ports.
+
+        :param if_node: The If node to update output ports and shapes
+        :return: None
+        """
+
+        then_outputs = [node for node in if_node.then_graph.get_op_nodes() if node.has('output_id')]
+        else_outputs = [node for node in if_node.else_graph.get_op_nodes() if node.has('output_id')]
+        outputs_mapping = {}
+        outputs_number = len(if_node.out_ports())
+
+        if outputs_number == 0 and len(if_node.out_ports(True)) != 0:
+            # Some models have if with control flow outputs.
+            # These shape inference for such ifs
+            # TODO: need to rethink and redo support for control flow edges in if operation
+            for node in if_node.out_nodes(True).values():
+                node.shape = int64_array([])
```

Comment:
please re-write to new Graph API:
            for output_port in if_node.out_ports(control_flow=True).values():
                output_port.data.set_shape(int64_array([]))

Reply:
Need to check center_net model, I remember there were troubles with new api